src/data/nat/prime.lean

```diff
@@ -491,6 +491,25 @@ by induction n with n IH;
    [exact pp.not_dvd_one.elim h,
     by { rw pow_succ at h, exact (pp.dvd_mul.1 h).elim id IH } ]
 
+lemma prime.pow_dvd_of_dvd_mul_right {p n a b : ℕ} (hp : p.prime) (h : p ^ n ∣ a * b)
+  (hpb : ¬ p ∣ b) : p ^ n ∣ a :=
+begin
+  induction n with n ih,
+  { simp },
+  { rw [pow_succ'] at *,
+    rcases ih (dvd_trans (dvd_mul_right _ _) h) with ⟨c, rfl⟩,
+    rw [mul_assoc] at h,
+    rcases hp.dvd_mul.1 (nat.dvd_of_mul_dvd_mul_left (pow_pos hp.pos _) h)
+      with ⟨d, rfl⟩|⟨d, rfl⟩,
+    { rw [← mul_assoc],
+      exact dvd_mul_right _ _ },
+    { exact (hpb (dvd_mul_right _ _)).elim } }
+end
+
+lemma _root_.nat.prime.pow_dvd_of_dvd_mul_left {p n a b : ℕ} (hp : p.prime) (h : p ^ n ∣ a * b)
```

Comment:
Why aren't those two lemmas in the same namespace?

Reply:
Because I made a mistake. Fixed.